Makefile

```diff
@@ -33,7 +33,7 @@ all:
 
 .PHONY: deps
 deps:
-	dep ensure -vendor-only
+	go mod download
```

Comment:
Related to the previous `-vendor-only` option, we need to ensure that the command here does not modify the go.mod/go.sum files.

Also, we need to ensure that the commands in other targets never download or modify the go.mod/go.sum.

Reply:
Looks like `go mod download` adhears to the current `go.sum` contents. Plus we do a `go mod verify`:
```
The 'go mod verify' command checks that
the cached copies of module downloads still match both their recorded
checksums and the entries in go.sum.
```

Reply:
We do this in circleci after the module download.

Reply:
If someone added a plugin with a new dependency but didn't add it to go.sum would this still pass?  The cached copies would still match the go.sum.

Reply:
Good though. Will experiment and check back in!

Reply:
I cannot find an equivalent to dep ensure --vendor-only. All the go mod commands appear to manipulate the `go.sum` file so I am adding a manual check using git diff.

Reply:
What about `-mod=readonly`?

https://github.com/golang/go/wiki/Modules#can-i-control-when-gomod-gets-updated-and-when-the-go-tools-use-the-network-to-satisfy-dependencies

Reply:
🙌 ahh great find

Reply:
Just had a play and basically you can put `-mod=readonly` on any non `mod` command. i.e. `go build` or `go install`. We would need to add that to everyone. The result of which is that command will fail iff any dependencies in the mod file are not in the build cache. However, if you do a `go mod download` they will already be there. At which point it still update the go.sum file with the one it has in its build cache. So the result is still an updated go.sum file that is not checked in.

Reply:
As far as I know  we use `go mod tidy` in order to assert that the `go.mod` and `go.sum` are in sync with the dependencies that the codebase really uses.

We have a script that checks that `go mod tidy` didn't produce any change in mod/sum files: https://github.com/influxdata/flux/blob/master/etc/checktidy.sh. That script is used  in CI in the step `make checktidy`: https://github.com/influxdata/flux/blob/master/Makefile#L58.

@danielnelson do you think that approach could solve your problem?